File: ComplaintService/src/main/java/com/complaints/service/ComplaintService.java
package com.complaints.service;

import com.complaints.entities.CommentEntity;
import com.complaints.entities.ComplaintEntity;
import com.complaints.response.ServiceResponse;

public interface ComplaintService {
    ServiceResponse createComplaint(ComplaintEntity complaintEntity);

	ServiceResponse getComplaint(String id);

    ServiceResponse createComment(CommentEntity commentEntity);
}

File: ComplaintMobile/WebApp/complaints/complaint/js/ComplaintService.js
angular.module("complaints.services").factory("ComplaintService", ['$http', function($http) {
	return {
		getComplaint: function(id) {
    		return $http.get('http://localhost:8080/ComplaintService/service/complaint/getComplaint/'+id).then(function(response) {
    			return response.data;
    		});
        },
        submitComment: function(comment) {
        	return $http.post('http://localhost:8080/ComplaintService/service/complaint/createComment',comment).then(function(response) {
    			return response.data;
    		});
        }
    };
}]);